bool leave
bool TZ1
bool TZ2
bool TZ3
